public/sw.js

```diff
@@ -107,11 +114,45 @@ self.addEventListener('fetch', (event) => {
 				}
 
 				// If the CSS/JS didn't change since it's been cached, return the cached version
-				if (hasHash(event.request.url) && hasSameHash(event.request.url, cached.url)) {
+				if (
+					hasHash(event.request.url)
+					&& hasSameHash(event.request.url, cached.url)
+				) {
 					return cached;
 				}
 			}
 			return fetchFromNetwork(event);
 		}),
 	);
 });
+
+self.addEventListener('push', function(event) {
+	const data = JSON.parse(event.data.text());
+
+	const options = {
+		body: data.message,
+		icon: data.icon,
+		vibrate: data.vibrate,
+		data: {
+			dateOfArrival: Date.now(),
+			redirectURL: data.redirectURL,
+		},
+		actions: [
+			{ action: 'reply', title: 'Reply' },
+			{ action: 'close', title: 'Close' },
+		],
+	};
+
+	event.waitUntil(self.registration.showNotification(data.title, options));
+});
+
+self.addEventListener('notificationclick', function(event) {
```

Comment:
We cannot use Meteor functions in service worker, so that was an issue with the reply feature. The konchat one's are handled by meteor while these are handled by service where only a limited functionality is available